package/MDAnalysis/core/groups.py

```diff
@@ -3324,6 +3324,56 @@ def write(self, filename=None, file_format=None,
 
         raise ValueError("No writer found for format: {}".format(filename))
 
+    def sort(self, **keys):
+        """Return sorted Atomgroup.
+
+        Example
+        -------
+
+        .. code-block:: python
+
+            >>> import MDAnalysis as mda
+            >>> from MDAnalysisTests.datafiles import PDB_small
+            >>> u = mda.Universe(PDB_small)
+            >>> ag = sum([u.atoms[3], u.atoms[2], u.atoms[1], u.atoms[0]])
+            >>> ag.ids
+            array([4 3 2 1])
+            >>> ag.ix
+            array([3 2 1 0])
+            >>> ag = ag.sort(key=lambda atom: atom.id)
+            >>> ag.ids
+            array([1 2 3 4])
+            >>> ag.ix
+            array([0 1 2 3])
+
+        Parameters
+        ----------
+        keys: callable object
+        A function for the parameter of each atom to use for sorting.
+        Example: lambda atom: atom.id
+
+        Returns
+        -------
+        output:
+            Sorted atomgroup.
+
+
+        .. versionadded:: 2.0.0
+        """
+        if len(keys) == 0:
+            raise NameError("argument 'key' is not defined. ")
+        for key, value in keys.items():
+            if key == "key":
+                n = len(self.atoms)
+                idx = np.zeros(n)
+                for i in range(n):
+                    idx[i] = value(self.atoms[i])
```

Comment:
I don't like the pattern of providing a lambda that is applied to each atom.  Mostly because it will scale poorly and AtomGroups are often large.  I think a better approach would be to take the name of an attribute (eg `"ids"`, `"masses"`) and have a `getattr` call to build the `idx` array here.  This can then still use the argsort afterwards.